packages/bff/src/services/privacyNoticeStorage.ts

```diff
@@ -0,0 +1,96 @@
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+
+import {
+  DynamoDBClient,
+  GetItemCommand,
+  PutItemCommand,
+  QueryCommand,
+} from "@aws-sdk/client-dynamodb";
+import { marshall, unmarshall } from "@aws-sdk/util-dynamodb";
+import { Logger } from "pagopa-interop-commons";
+import { dynamoReadingError } from "../model/domain/errors.js";
+import { PrivacyNotice, UserPrivacyNotice } from "../model/domain/types.js";
+
+export function privacyNoticeStorageServiceBuilder(
+  db: DynamoDBClient,
+  privacyNoticesTableName: string,
+  privacyNoticesUsersTableName: string
+) {
+  return {
+    async getLatestVersion(
+      id: string,
+      logger: Logger
+    ): Promise<PrivacyNotice | null> {
+      logger.info(`Getting id ${id} privacy notice`);
+
+      try {
+        const command = new GetItemCommand({
+          TableName: privacyNoticesTableName,
+          Key: { privacyNoticeId: { S: id } },
+        });
+        const result = await db.send(command);
+
+        if (result.Item === undefined) {
+          return null;
+        }
+        return unmarshall(result.Item) as PrivacyNotice;
+      } catch (error) {
+        throw dynamoReadingError(error);
+      }
+    },
+
+    async getByUserId(
+      id: string,
+      userId: string,
+      logger: Logger
+    ): Promise<UserPrivacyNotice | null> {
+      logger.info(`Getting privacy notice with id ${id} for user ${userId}`);
+
+      try {
+        const command = new QueryCommand({
+          TableName: privacyNoticesUsersTableName,
+          KeyConditionExpression: "pnIdWithUserId = :pnIdWithUserId",
+          ExpressionAttributeValues: {
+            ":pnIdWithUserId": { S: `${id}#${userId}` },
+          },
+        });
+
+        const result = await db.send(command);
+        if (!result.Items || result.Items.length === 0) {
+          return null;
+        }
+
+        const items = result.Items.map(
+          (item) => unmarshall(item) as UserPrivacyNotice
```

Comment:
Maybe instead of using `as` we could create some zod decoders (but maybe it's unnecessary). @Viktor-K @ecamellini @sandrotaje what do you think?

Reply:
Given the data comes from an external data source, I would also perform parsing with `zod`.
Let's re-define the types as `zod` types and here use `.safeParse`, as we do for readmodel reads, so that we have data parsing & validation and we can throw an error here in case of parsing issues

Reply:
I think that is a great idea, i don't like to use `as` either especially for external data. 
I did implement this using zod and `safeParse`, if you wish you can take a look and tell me if you like it.